File: Services/UnionService/UnionService.Application/UseCases/InvitationUseCases/AddInvitation/AddInvitationCommand.cs
using MediatR;
using UnionService.Domain.Enums;

namespace UnionService.Application.UseCases
{
    public record AddInvitationCommand(
        string? username,
        int? companyId,
        string? memberUsername,
        string? description,
        CompanyRole? role,
        int? teamId,
        InvitationType? type
    ) : IRequest<Unit>;
}

File: FrontendApp/src/app/models/commands/unionservice/add-invitation.command.ts
import { CompanyRole } from "../../../enums/company-role.enum";
import { InvitationType } from "../../../enums/invitation-type.enum";

export interface AddInvitationCommand {
  username: string | null;
  companyId: number | null;
  memberUsername: string | null;
  description: string | null;
  role: CompanyRole | null;
  teamId: number | null;
  type: InvitationType | null;
}
